src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,888 @@
+        if model.MD_LENS_NAME in self._metadata:
+            optional_data.append(OptionalDataRecord("Objective", PQ_OPT_DATATYPE_FIXED_LENGTH_STRING,
+                                        self._metadata[model.MD_LENS_NAME]))
+
+        logging.info("Requesting an acquisition. Pixel size: %f m, Resolution: %d x %d", pixel_size, iPixelNumber_X, iPixelNumber_Y)
+
+        msg = DataframeServerRequestMessage(T_REC_VERSION, PQ_MEASTYPE_IMAGESCAN,
+                                iPixelNumber_X, iPixelNumber_Y, 
+                                iScanningPattern, pixel_size, 
+                                optional_data)
+
+        self._sendMessage(msg)
+
+        self._tStart = time.time()  # the measuremnet start time in seconds from the epoch
+        self._measurinement_stopped.clear()
+
+    def StopMeasurement(self):
+        '''
+        Sends a user break to the server to stop the measurement. 
+        '''
+        if self.isMeasuring():
+            logging.info("Sending user break.")
+            self._sendMessage(EncodedStatusMessage (PQ_STOPREASON_CODE_USER_BREAK))
+            self._user_break = True
+
+    def _listen(self):
+        '''
+        This method runs in a separate thread and listens for messages sent to
+        the device via IP sockets
+        '''
+        logging.info("Starting listening thread...")
+        msg = ''
+        try:
+            while not self._shutdown_flag:
+
+                # This will black, but timeout if no message is received in 2 s
+                try:
+                    msg += self._socket.recv(4096)
+                except socket.timeout:
+                    # this is ok. Just means the server didn't send anything.
+                    # Keep listening
+                    pass
+
+                if len(msg) < 2:
+                    # determine if we got enough data to determine how much we need
+                    continue
+                
+                # once we have two characters, we can read the message length
+                msg_len = struct.unpack_from('H', msg)[0]
+
+                # keep receiving data until we have at least one message
+                if len(msg) < msg_len:
+                    continue
+
+                decoded_msg = Message.from_bytes(msg)
+                logging.debug("R: Msgtype %s", decoded_msg)
+
+                try:
+                    self._actOnMessage(decoded_msg)
+                except HwError as e:
+                    # set the component state from a hardware error.
+                    logging.error(e)
+                    self.state._set_value(e, force_write=True)
+                except Exception as e:
+                    # any other type of exception.
+                    logging.error(e)
+                finally:
+                    # reset message buffer to receive the next.
+                    msg = ''
+
+        except Exception as e:
+            # another exception. End running.
+            logging.error(e)
```

Comment:
You should use `logging.exception`, which does the same as `logging.error`, but also prints a backtrace.